.github/shared/src/swagger.js

```diff
@@ -132,6 +193,23 @@ export class Swagger {
     return this.#tag;
   }
 
+  /**
+   * @returns {string} version kind (stable or preview)
+   */
+  get versionKind() {
+    return dirname(this.#path).includes("/preview/")
+      ? API_VERSION_LIFECYCLE_STAGES.PREVIEW
+      : API_VERSION_LIFECYCLE_STAGES.STABLE;
+  }
+
+  /**
+   * @returns {Promise<string>} file name of the swagger file
+   */
+  async getFileName() {
```

Comment:
Typically, the "file name" of a path is everything after the last folder.  However, below you say this includes everything after the version dir, so it may include folders like "types/common.js".  So, I think this should either be renamed, or moved out of spec-model.  Why or how would this be useful to other consumers of spec-model?